UnitTests/GitCommandsTests/CommitTemplateManagerTests.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using FluentAssertions;
+using GitCommands;
+using GitUIPluginInterfaces;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace GitCommandsTests
+{
+    [TestFixture]
+    public class CommitTemplateManagerTests
+    {
+        private IGitModule _module;
+        private FileBase _file;
+        private IFileSystem _fileSystem;
+        private CommitTemplateManager _manager;
+
+
+        [SetUp]
+        public void Setup()
+        {
+            _module = Substitute.For<IGitModule>();
+            _file = Substitute.For<FileBase>();
+            _fileSystem = Substitute.For<IFileSystem>();
+            _fileSystem.File.Returns(_file);
+            _manager = new CommitTemplateManager(_module, _fileSystem);
+        }
+
+
+        [TestCase(null)]
+        [TestCase("")]
+        public void LoadGitCommitTemplate_commit_template_not_defined(string template)
+        {
+            _module.GetEffectiveSetting("commit.template").Returns(template);
+
+            _manager.LoadGitCommitTemplate().Should().BeNull();
+        }
+
+        [TestCase("~/template.txt")]
+        [TestCase("./template.txt")]
+        [TestCase("template.txt")]
+        public void LoadGitCommitTemplate_path_not_rooted(string template)
+        {
+            const string workingDir = @"c:\dev\repo";
+            _module.WorkingDir.Returns(workingDir);
+            _module.GetEffectiveSetting("commit.template").Returns(template);
+
+            ((Action)(() => _manager.LoadGitCommitTemplate())).ShouldThrow<FileNotFoundException>();
+        }
+
+        [Test]
+        public void LoadGitCommitTemplate_should_load_file_content()
```

Comment:
I see here 3 tests combined:
1. LoadGitCommitTemplate_ExpandsRelativePathToWorkingDirSubpath
2. LoadGitCommitTemplate_should_load_file_content
3. LoadGitCommitTemplate_RmovesCRFormFileContent

Reply:
Yes, if you insist on nitpicking :)

Reply:
No, I am only a human, so do other devs and we all make mistakes. When there are no explicit assumptions given, then it is hard to reason if the test author wrote the test correctly (if the test implementation is correct according to the spec).